### ISSUE TITLE
Failure to change DynamoDB billing_mode from PAY_PER_REQUEST to PROVISIONED

### ISSUE THREAD

**Issue by nathmclean** [terraform-providers/terraform-provider-aws#7097]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.3

### Affected Resource(s)

* aws_dynamodb_table

### Terraform Configuration Files

```hcl
provider "aws" {
  region              = "us-east-1"
}

resource "aws_dynamodb_table" "test" {
  name = "on_demand_test_nathan"
#  billing_mode   = "PROVISIONED"   
  billing_mode   = "PAY_PER_REQUEST"
  read_capacity  = 20
  write_capacity = 20
  hash_key       = "UserId"
  range_key      = "GameTitle"

  attribute {
    name = "UserId"
    type = "S"
  }

  attribute {
    name = "GameTitle"
    type = "S"
  }

  tags = {
      SAG_PRODUCT = "test"
  }

  lifecycle {
    prevent_destroy = true
  }
}
```

### Debug Output

https://gist.github.com/nathmclean/61ac0b1fbd69f84bce6d9cf4cbfdbf9c

### Expected Behavior

The dynamo Table should be changed from a billing_mode of PAY_PER_REQUEST to PROVISIONED

### Actual Behavior

The Terraform run fails as the ProvisionedThroughput is not included in the request

```
Error: Error applying plan:

1 error(s) occurred:

* aws_dynamodb_table.test: 1 error(s) occurred:

* aws_dynamodb_table.test: Error updating DynamoDB Table (on_demand_test_nathan) billing mode: ValidationException: One or more parameter values were invalid: ProvisionedThroughput must be specified when BillingMode is PROVISIONED
        status code: 400, request id: 3VHI1DAP29B7K2C0LJD8V69MN7VV4KQNSO5AEMVJF66Q9ASUAAJG

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

### Steps to Reproduce

1. Change `billing_mode` in the HCL above to PAY_PER_REQUEST
2. `terraform apply`
3.  Change `billing_mode` in the HCL above to PROVISIONED
4. `terraform apply`

### References

* #0000

**Comment by aroder** [comment]
I had this error until I upgraded the AWS provider. Try running `terraform init --upgrade`. Now I have 1.56.0 and I no longer receive the error.

**Comment by nathmclean** [comment]
Just upgraded (1.57.0) and received the same error as initially reported

**Comment by aroder** [comment]
Do you get the same error when creating a new one with billing mode PAY_PER_REQUEST, or only when attempting to switch billing modes?

**Comment by nathmclean** [comment]
Only when attempting to switch from PAY_PER_REQUEST to PROVISIONED. I'm just prepping a PR with a fix. The capacity isn't being included in the request when updating billing_mode

**Comment by plektra** [comment]
@nathmclean your last comment is about changing from PAY_PER_REQUEST to PROVISIONED, I'm having the same issue. The title of the original post was actually about changing the billing mode other way around (provisioned to on-demand) so I'm a bit confused what's the case here. :)

Nevertheless, my issue seems to be related to global secondary index, I'm getting following error when updating the billing mode of a table with one or more GSIs:
```2019-01-31T11:49:36.219+0200 [DEBUG] plugin.terraform-provider-aws_v1.51.0_x4: 2019/01/31 11:49:36 [DEBUG] [aws-sdk-go] {"__type":"com.amazon.coral.validate#ValidationException","message":"One or more parameter values were invalid: ProvisionedThroughput must be specified for index: username"}```

What seems to be happening is that Terraform trying to create a new index for some reason, and the existing index is left with "0" as the provisioned capacity while new index is being applied with the proper capacity values.

```2019/01/31 11:59:50 aws_dynamodb_table.users: Modifying... (ID: users)
  billing_mode:                                           "PAY_PER_REQUEST" => "PROVISIONED"
  global_secondary_index.3596640839.hash_key:             "" => "nick_lower"
  global_secondary_index.3596640839.name:                 "" => "username"
  global_secondary_index.3596640839.non_key_attributes.#: "0" => "1"
  global_secondary_index.3596640839.non_key_attributes.0: "" => "nick"
  global_secondary_index.3596640839.projection_type:      "" => "INCLUDE"
  global_secondary_index.3596640839.range_key:            "" => "discriminator"
  global_secondary_index.3596640839.read_capacity:        "" => "40"
  global_secondary_index.3596640839.write_capacity:       "" => "1"
  global_secondary_index.3667214579.hash_key:             "nick_lower" => ""
  global_secondary_index.3667214579.name:                 "username" => ""
  global_secondary_index.3667214579.non_key_attributes.#: "1" => "0"
  global_secondary_index.3667214579.non_key_attributes.0: "nick" => ""
  global_secondary_index.3667214579.projection_type:      "INCLUDE" => ""
  global_secondary_index.3667214579.range_key:            "discriminator" => ""
  global_secondary_index.3667214579.read_capacity:        "0" => "0"
  global_secondary_index.3667214579.write_capacity:       "0" => "0"
  read_capacity:                                          "0" => "1"
  write_capacity:                                         "0" => "5"```

What I'm expecting is the index to be left intact besides the capacity.

**Comment by nathmclean** [comment]
@plektra Sorry about that, I've fixed the title.

Certainly looks like it's creating a new table and trying to keep both - should just modify the existing GSI. Not sure I'll get chance to update the PR to include tests and fixes for this though.

**Comment by bflad** [comment]
The fix for the error when updating `billing_mode` from `PAY_PER_REQUEST` to `PROVISIONED` and an accompanying acceptance test has been merged and will release with version 1.58.0 of the Terraform AWS provider, likely middle of this week.

@plektra if your issue isn't fixed by #6737, which was released in version 1.52.0 of the Terraform AWS provider, I would suggest opening a new issue so we can appropriately triage. 👍

**Comment by plektra** [comment]
@bflad I'm using AWS provider version 1.57.0 and the issue exists, so I'll check if this issue is already reported and will open a new one if not. Thanks!

**Comment by bflad** [comment]
This has been released in [version 1.58.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1580-february-08-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
